File: omnipod-dash/src/test/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/pod/response/DefaultStatusResponseTest.java
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.response;

import org.apache.commons.codec.DecoderException;
import org.apache.commons.codec.binary.Hex;
import org.junit.Test;

import info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.definition.DeliveryStatus;
import info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.definition.PodStatus;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotSame;

public class DefaultStatusResponseTest {
    @Test
    public void testValidResponse() throws DecoderException {
        byte[] encoded = Hex.decodeHex("1D1800A02800000463FF");
        DefaultStatusResponse response = new DefaultStatusResponse(encoded);

        assertArrayEquals(encoded, response.getEncoded());
        assertNotSame(encoded, response.getEncoded());
        assertEquals(ResponseType.DEFAULT_STATUS_RESPONSE, response.getResponseType());
        assertEquals(ResponseType.DEFAULT_STATUS_RESPONSE.getValue(), response.getMessageType());
        assertEquals(DeliveryStatus.BASAL_ACTIVE, response.getDeliveryStatus());
        assertEquals(PodStatus.RUNNING_ABOVE_MIN_VOLUME, response.getPodStatus());
        assertEquals((short) 320, response.getTotalPulsesDelivered());
        assertEquals((short) 5, response.getSequenceNumberOfLastProgrammingCommand());
        assertEquals((short) 0, response.getBolusPulsesRemaining());
        assertFalse(response.isOcclusionAlertActive());
        assertFalse(response.isAlert1Active());
        assertFalse(response.isAlert2Active());
        assertFalse(response.isAlert3Active());
        assertFalse(response.isAlert4Active());
        assertFalse(response.isAlert5Active());
        assertFalse(response.isAlert6Active());
        assertFalse(response.isAlert7Active());
        assertEquals((short) 280, response.getMinutesSinceActivation());
        assertEquals((short) 1023, response.getReservoirPulsesRemaining());
    }
}

File: omnipod-dash/src/test/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/pod/response/DefaultStatusResponseTest.kt
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.response

import info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.definition.DeliveryStatus
import info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.definition.PodStatus
import org.apache.commons.codec.DecoderException
import org.apache.commons.codec.binary.Hex
import org.junit.Assert
import org.junit.Test

class DefaultStatusResponseTest {

    @Test @Throws(DecoderException::class) fun testValidResponse() {
        val encoded = Hex.decodeHex("1D1800A02800000463FF")
        val response = DefaultStatusResponse(encoded)
        Assert.assertArrayEquals(encoded, response.encoded)
        Assert.assertNotSame(encoded, response.encoded)
        Assert.assertEquals(ResponseType.DEFAULT_STATUS_RESPONSE, response.responseType)
        Assert.assertEquals(ResponseType.DEFAULT_STATUS_RESPONSE.value, response.messageType)
        Assert.assertEquals(DeliveryStatus.BASAL_ACTIVE, response.deliveryStatus)
        Assert.assertEquals(PodStatus.RUNNING_ABOVE_MIN_VOLUME, response.podStatus)
        Assert.assertEquals(320.toShort(), response.totalPulsesDelivered)
        Assert.assertEquals(5.toShort(), response.sequenceNumberOfLastProgrammingCommand)
        Assert.assertEquals(0.toShort(), response.bolusPulsesRemaining)
        Assert.assertEquals(0, response.activeAlerts.size)
        Assert.assertEquals(280.toShort(), response.minutesSinceActivation)
        Assert.assertEquals(1023.toShort(), response.reservoirPulsesRemaining)
    }
}